examples/src/main/scala/org/apache/spark/examples/mllib/PowerIterationClusteringExample.scala

```diff
@@ -0,0 +1,176 @@
+ *   k:  Number of circles/ clusters
+ *   n:  Total number of sampled points. There are proportionally more points within the
+ *      outer/larger circles
+ *   numIterations:   Number of Power Iterations
+ *   outerRadius:  radius of the outermost of the concentric circles
+ * }}}
+ *
+ * Here is a sample run and output:
+ *
+ * ./bin/run-example org.apache.spark.examples.mllib.PowerIterationClusteringExample -k 3 --n 30 --numIterations 15
+ *
+ * Cluster assignments: 1 -> [0,1,2,3,4],2 -> [5,6,7,8,9,10,11,12,13,14],
+ *    0 -> [15,16,17,18,19,20,21,22,23,24,25,26,27,28,29]
+ *
+ *
+ * If you use it as a template to create your own app, please use `spark-submit` to submit your app.
+ */
+object PowerIterationClusteringExample {
+
+  case class Params(
+      input: String = null,
+      k: Int = 3,
+      numPoints: Int = 30,
+      numIterations: Int = 10,
+      outerRadius: Double = 3.0
+      ) extends AbstractParams[Params]
+
+  def main(args: Array[String]) {
+    val defaultParams = Params()
+
+    val parser = new OptionParser[Params]("PIC Circles") {
+      head("PowerIterationClusteringExample: an example PIC app using concentric circles.")
+      opt[Int]('k', "k")
+        .text(s"number of circles (/clusters), default: ${defaultParams.k}")
+        .action((x, c) => c.copy(k = x))
+      opt[Int]('n', "n")
+        .text(s"number of points, default: ${defaultParams.numPoints}")
+        .action((x, c) => c.copy(numPoints = x))
+      opt[Int]("numIterations")
+        .text(s"number of iterations, default: ${defaultParams.numIterations}")
+        .action((x, c) => c.copy(numIterations = x))
+    }
+
+    parser.parse(args, defaultParams).map { params =>
+      run(params)
+    }.getOrElse {
+      sys.exit(1)
+    }
+  }
+
+  def generateCircle(n: Int, r: Double): Array[(Double, Double)] = {
+    val pi2 = 2 * math.Pi
+    (0.0 until pi2 by pi2 / n).map { x =>
+      (r * math.cos(x), r * math.sin(x))
+    }.toArray
+  }
+
+  def generateCirclesRdd(sc: SparkContext, nCircles: Int = 3, nTotalPoints: Int = 30,
+                         outerRadius: Double):
+  RDD[(Long, Long, Double)] = {
+    // The circles are generated as follows:
+    // The Radii are equal to the largestRadius/(C - circleIndex)
+    //   where  C=Number of circles
+    //      and the circleIndex is 0 for the innermost and (nCircles-1) for the outermost circle
+    // The number of points in each circle (and thus in each final cluster) is:
+    //     x, 2x, .., nCircles*x
+    //     Where x is found from  x = N * C(C+1)/2
+    //  The # points in the LAST circle is adjusted downwards so that the total sum is equal
+    //  to the nTotalPoints
+
+    val smallestRad = math.ceil(nTotalPoints / (nCircles * (nCircles + 1) / 2.0))
+    var groupSizes = (1 to nCircles).map(gs => (gs * smallestRad).toInt)
+    groupSizes.zipWithIndex.map { case (gs, ix) =>
+      ix match {
+        case _ if ix == groupSizes.length => gs - (groupSizes.sum - nTotalPoints)
```

Comment:
This branch is because that we allow users to set `n`. This can be simplified if we only allow users to set the size of the smallest circle. Then the code becomes

~~~
val groupSizes = Array.tabulate(nCircles).map(i => n * (i + 1))
~~~